makeflow/src/makeflow_catalog_reporter.c

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright (C) 2016- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include "dag.h"
+#include "timestamp.h"
+#include "stringtools.h"
+#include "list.h"
+#include "makeflow_summary.h"
+#include "catalog_query.h"
+#include "json.h"
+#include "json_aux.h"
+#include "username.h"
+#include "batch_job.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <string.h>
+
+//packet for start and stop, once per min, configurable time
+
+//make sure to send: project name, owner, type
+
+//waiting means: inputs not ready, running means: took job and submitted
+
+int makeflow_catalog_summary(struct dag* d, char* name, batch_queue_type_t type, timestamp_t start){
+    struct dag_node *n;
+    dag_node_state_t state;
+    
+    int tasks_completed = 0;
+    int tasks_aborted   = 0;
+    int tasks_waiting   = 0;
+    int tasks_running   = 0;
+    int tasks_failed    = 0;
+
+    for (n = d->nodes; n; n = n->next) {
+        state = n->state;
+        if (state == DAG_NODE_STATE_FAILED)
+            tasks_failed++;
+        else if (state == DAG_NODE_STATE_ABORTED)
+            tasks_aborted++;
+        else if (state == DAG_NODE_STATE_COMPLETE) 
+            tasks_completed++;
+        else if(state == DAG_NODE_STATE_RUNNING)
+            tasks_running++;
+        else if(state == DAG_NODE_STATE_WAITING)
+            tasks_waiting++;
+    }
+    
+    //transmit report here
+    //creates memory
+    char* host = string_format("%s:%i",CATALOG_HOST,CATALOG_PORT);
```

Comment:
There's no need to add the port yourself. Since we have two servers in the default `CATALOG_HOST`, this actually just specifies a port for the second one. `CATALOG_PORT` is somewhat deprecated; it's just the default port to use if none is specified in one of the entries in `CATALOG_HOST`. You can just pass `CATALOG_HOST` directly to `catalog_port_send_update`, it'll figure out the ports.